webapp/channels/src/components/file_attachment_list/index.ts

```diff
@@ -24,14 +25,24 @@ export type OwnProps = {
     compactDisplay?: boolean;
     isInPermalink?: boolean;
     handleFileDropdownOpened?: (open: boolean) => void;
+    isEditHistory?: boolean;
+    disableDownload?: boolean;
+    disableActions?: boolean;
```

Comment:
A couple thoughts:
1. Why can't these be derived from `fileInfo.delete_at` like `disableThumbnail` and `disablePreview` are? I like automatically doing that since it means that we only have fewer possible combinations of props for this component and fewer possible cases to test.
2. Along those lines, if we can't derive those because this just ties into whether or not this is being used for the edit history, I think it'd be better to group `disablePreview`, `disableDownload`, and `disableActions` into a single property (either `disableActions` or `isEditHistory` would work) so that we have fewer possible cases to test for how this displays. The more granular settings would be nice if we ever were to use this component in more places with more customization, but that'll make it harder to maintain, so I'd lean towards fewer options until we find we need them.

Reply:
Because I want to disable them in edit history even if the file isn't deleted.

2nd point is subjective and I've received contradictory review comments from different reviewers in the past multiple times on similar changes. `isEditHistory` prop can be used to club the three but it becomes unnecessarily tied to edit history when there is no reason for it to. I now prefer to keep things general and agnostic to a feature. Thus I'm skipping this change.

Reply:
1 makes sense, and yeah, I was aware that my second point is a bit contradictory to even my own reviews from before, so I'm good keeping this as-is.

I've been trying to think of what the ideal way to share functionality/appearance of components like this is without having a mess of conditional logic in them, but I haven't been able to test out any of my ideas. Currently, I'm torn between abstracting that logic out like I pitched with the `EditorContext` in the `AdvancedTextEditor` and breaking up the component more so that so that we could theoretically have a `PostFileAttachmentList` and an `EditHistoryFileAttachmentList` which both use a dumb `FileAttachmentList` with minimal logic under the hood other than rendering. Either way, I think we'd need to remove some or all of the Redux logic from the component. I wouldn't be surprised if there isn't a single solution for all cases though given they all work differently